tests/api/metrics_test.py

```diff
@@ -39,26 +39,26 @@ def test_read_metric(session: Session, client: TestClient) -> None:
                 database=Database(name="test", URI="sqlite://"),
                 table="A",
                 columns=[
-                    Column(name="ds", type=ColumnType.STR),
-                    Column(name="user_id", type=ColumnType.INT),
-                    Column(name="foo", type=ColumnType.FLOAT),
+                    Column(name="ds", type=StringType()),
```

Comment:
I think the issue is that it's harder to handle more complex types with the old type system. For example, `DecimalType` has a precision and scale, but that's harder to store on the old system since it was just represented as a string. @CircArgs any additional thoughts?

Reply:
In short, you should always use it with parentheses. This is how Iceberg ended up doing it. I had tried to do something like `String` but to be similar to the Java implementation we went with singletons like this. We could namespace them under `ColumnType` since they're singletons, but I'm not sure it's meaningful for our internal development. Another alternative is to just break out the parsing logic you can find in the visitors which would give us the same thing as the old metaclass stuff. To Yian's point, we do take arguments for some types naturally and rather than using generics with literals, we use singletons pretty naturally I think.